### DISCUSSION TITLE
What is the startup speed for normcap?

### DISCUSSION THREAD

**Discussion by Norlandz** [dynobo/normcap#678] [Q&A] [answered]
Is there **any startup speed difference among**
1. click the already running normcap icon in tray icon 
2. double click and run the exe (regardless of already running or not)
3. run `"C:\Program Files\NormCap\NormCap.exe"` in terminal (regardless of already running in the terminal or background or not)

In practice, it seems no difference, **all takes 1s,** measured by eye.

But theoretically, shouldnt there be a speed difference between "normcap is already running" vs "normcap is not already running (**cold start**)"?

(
This post is created regarding the discussion of hotkey

Enhancement Suggestion - Add Customizable Hotkeys (Keyboard Shortcuts) · Issue #590 · dynobo/normcap

normcap/adr/004-do-not-implement-hotkey.md at main · dynobo/normcap
https://github.com/dynobo/normcap/blob/main/adr/004-do-not-implement-hotkey.md
)

(Im currently using terminal + AHK to run Normcap.)

**Comment by dynobo** [comment, marked as answer]
**TLDR; Capturing with NormCap from tray icon is faster compared to a fresh start, BUT the difference is so small, it barely matters in practice.**

NormCap was always designed to start fast to not require running it as daemon in tray. NormCap barely needs to load state (mainly the application settings), so there is little to win by running in tray. But, as every Python application, there is _always_ a startup overhead for loading & running the Python interpreter and importing all modules. _How much_ time this takes depends on your system, last time I measured it on my machine, it was like 300ms. 

The processing time for text recognition also depends on the system, but mostly on the selected region's pixel size. On my system, even for a really small region, it takes 1-2 seconds.  

That means, the startup delay is _always_ quite small compared to the rest of the processing time. For most use cases, I would recommend to _not_ use the tray.

Possible reasons for _still_ using the tray might be:
- On slow systems, when startup delay is noticeable long.
- Depending on the workflow for usability reasons, e.g. for mouse driven people.
- If you need to do _a lot_ of consecutive captures. (But if you are doing this, I'd say you should consider different tooling/workflow anyway)

Does this answer your questions? :slightly_smiling_face:

**Reply by KamilDev** [reply]
I do a lot of OCR snipping, so I wouldn't mind keeping in the tray for a tiny faster launch.

However, I configured it to launch the program with a hotkey. Is there any way to make hotkey work with the tray? After being in tray, it no longer gets invoked by hotkey (launch exe). Is there any args or terminal command I can bind to the hotkey instead to make it compatible with using the tray?
